icgcget/cli.py

```diff
@@ -112,90 +113,93 @@ def cli(ctx, config, logfile):
 @click.option('--icgc-transport-file-from')
 @click.option('--icgc-transport-memory')
 @click.option('--icgc-transport-parallel')
-def download(repo, fileid, manifest, output,
+def download(repo, fileids, manifest, output,
              cghub_access, cghub_path, cghub_transport_parallel,
              ega_access, ega_password, ega_path, ega_transport_parallel, ega_udt, ega_username,
              gdc_access, gdc_path, gdc_transport_parallel, gdc_udt,
              icgc_access, icgc_path, icgc_transport_file_from, icgc_transport_memory, icgc_transport_parallel):
     logger = logging.getLogger('__log__')
-    code = 0
 
-    if fileid is None and manifest is None:
-        logger.error("Please provide either a file id value or a manifest file to download.")
-        code = 1
-        return code
-
-    if repo == 'ega':
+    object_ids = {}
+    if 'ega' in repo:
         if ega_username is None or ega_password is None:
-            if ega_access is None:
+            if ega_access is None:  # Do we want to raise exceptions on these, or just run them all?
                 logger.error("No credentials provided for the ega repository.")
-                code = 1
-                return code
-        if fileid is not None:
-            if len(fileid) > 1:
+                raise click.BadParameter
+    if 'collaboratory' in repo or 'aws-virginia' in repo:
+        if icgc_access is None:
+            logger.error("No credentials provided for the icgc repository")
+            raise click.BadParameter
+    if 'cghub' in repo:
+        if cghub_access is None:
+            logger.error("No credentials provided for the cghub repository.")
+            raise click.BadParameter
+    if 'gdc' in repo:
+        if gdc_access is None:
+            logger.error("No credentials provided for the gdc repository.")
+            raise click.BadParameter
+
+    for repository in repo:
+        object_ids[repository] = []
+
+    for fileid in fileids:
+        info = icgc_api.check_file(fileid)
+        if type(info) is int:
+            raise click.ClickException("File not found")
+        repository, copy = match_repositories(repo, info)
+        if repository is None:
+            logger.error("File {} specified not found on repositories {}".format(info["id"], repo))
+        object_ids[repository].append(copy["indexFile"]["objectId"])
+
+    if 'aws-virginia' in repo:
+        if 'aws-virginia' in object_ids and object_ids['aws-virginia']:
+            if len(object_ids['aws-virginia']) > 1:  # Todo-find a workaround for this: dynamic manifest generation?
+                logger.error("The icgc repository does not support input of multiple file id values.")
+                raise click.BadParameter
```

Comment:
The use won't see anything. I recommend adding a message to the exception always. It may not be as detailed as your log statement, and sometimes it may be exactly the same, but some context for the user goes a long way.